cloud/vmware/vmware_datacenter.py

```diff
@@ -74,18 +74,29 @@
     HAS_PYVMOMI = False
 
 
-def state_create_datacenter(module):
-    datacenter_name = module.params['datacenter_name']
-    content = module.params['content']
-    changed = True
-    datacenter = None
+def get_datacenter(context, module):
+    try:
+        datacenter_name = module.params.get('datacenter_name')
+        datacenter = find_datacenter_by_name(context, datacenter_name)
+        return datacenter
+    except vmodl.RuntimeFault as runtime_fault:
+        module.fail_json(msg=runtime_fault.msg)
+    except vmodl.MethodFault as method_fault:
+        module.fail_json(msg=method_fault.msg)
 
-    folder = content.rootFolder
+
+def create_datacenter(context, module):
+    datacenter_name = module.params.get('datacenter_name')
+    folder = context.rootFolder
+    changed = False
 
     try:
-        if not module.check_mode:
-            datacenter = folder.CreateDatacenter(name=datacenter_name)
-        module.exit_json(changed=changed, result=str(datacenter))
+        if not module.check_mode:     
```

Comment:
check mode condition does not make sense this early, it should be right before the CreateDatacenter.